packages/PlanLimitsUI/src/pages/Limits/useAppState.ts

```diff
@@ -0,0 +1,450 @@
+
+    setAppState({
+      ...appState,
+      council,
+      whaituaId,
+      whaitua,
+
+      // SW
+      surfaceWaterMgmtUnitId,
+      surfaceWaterMgmtUnitDescription,
+      surfaceWaterMgmtUnitLimit,
+      surfaceWaterMgmtUnitAllocated,
+      surfaceWaterMgmtUnitAllocatedPercentage,
+      surfaceWaterMgmtSubUnitId,
+      surfaceWaterMgmtSubUnitDescription,
+      surfaceWaterMgmtSubUnitLimit,
+      surfaceWaterMgmtSubUnitAllocated,
+      surfaceWaterMgmtSubUnitAllocatedPercentage,
+
+      swLimit,
+      // GW
+      groundWaterZoneName,
+      groundWaterZones,
+      gwLimits,
+      // Flow
+      site,
+      minimumFlowLimitId,
+      flowRestrictionsLevel,
+      flowRestrictionsManagementSiteName,
+      flowRestrictionsManagementSiteId,
+    });
+  };
+
+  return [appState, setAppStateFromResult];
+}
+
+const findFeature = (
+  features: mapboxgl.MapboxGeoJSONFeature[],
+  layer: string,
+  prop: string
+) =>
+  features.find((feat) => feat.layer.id === layer)?.properties?.[prop] as
+    | string
+    | undefined;
+
+const findFeatureId = (
+  features: mapboxgl.MapboxGeoJSONFeature[],
+  layer: string
+) => features.find((feat) => feat.layer.id === layer)?.id as string;
+
+interface SWLimit {
+  subUnitLimit?: string;
+  unitLimit?: string;
+  useDefaultRuleForUnit: boolean;
+  useDefaultRuleForSubUnit: boolean;
+  mergeUnit?: boolean;
+  mergeSubUnit?: boolean;
+}
+
+function getSwLimit(
+  whaituaId: string,
+  surfaceWaterMgmtUnitLimit?: string,
+  surfaceWaterMgmtSubUnitLimit?: string
+): SWLimit {
+  const stat = {
+    unitLimit: surfaceWaterMgmtUnitLimit,
+    subUnitLimit: surfaceWaterMgmtSubUnitLimit,
+    useDefaultRuleForUnit:
+      !surfaceWaterMgmtUnitLimit && !surfaceWaterMgmtSubUnitLimit,
+    useDefaultRuleForSubUnit:
+      (!surfaceWaterMgmtUnitLimit && !surfaceWaterMgmtSubUnitLimit) ||
+      (!surfaceWaterMgmtSubUnitLimit && whaituaId === '4'),
+  };
+  return stat;
+}
+
+interface GWLimit {
+  depth: string;
+  category?: string;
+  subUnitLimit?: string;
+  unitLimit?: string;
+  useDefaultRuleForUnit: boolean;
+  useDefaultRuleForSubUnit: boolean;
+  parentSWUnitId?: number;
+  parentSWSubUnitId?: number;
+  unitAllocated?: {
+    amount?: string;
+    percentage?: number;
+  };
+  subUnitAllocated?: {
+    amount?: string;
+    percentage?: number;
+  };
+  mergeUnit?: boolean;
+  mergeSubUnit?: boolean;
+}
+
+function getGwLimits(
+  activeFeatures: mapboxgl.MapboxGeoJSONFeature[],
+  surfaceWaterMgmtUnitLimit?: string
+) {
+  const rows: GWLimit[] = [];
+  if (activeFeatures.length === 0) {
+    rows.push({
+      depth: 'All',
+      useDefaultRuleForSubUnit: true,
+      useDefaultRuleForUnit: true,
+    });
+    return rows;
+  }
+
+  activeFeatures
+    .filter((feature) => feature.properties?.category === 'Category A')
+    .forEach((feature) => {
+      if (
+        !feature.properties?.surface_water_unit_allocation_amount &&
+        !surfaceWaterMgmtUnitLimit
+      ) {
```

Comment:
@smozely does this check seem right? It was copied over from compileLimits table, and I can't remember why it was added there. I'm not sure why we're checking for a _map based_ SW management unit limit (in addition to the one defined by the parent relationship).

Reply:
No, I don't think that should be there ... maybe a hangover. But I think we should only be using the information thats coming in via the `feature.properties`